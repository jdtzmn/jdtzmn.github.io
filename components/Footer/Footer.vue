<template lang="pug">
  .footer
    .footer-content
      .footer-section(v-for='section in sections', :key='section.name')
        .footer-header {{ section.name }}
        .footer-link(v-for='link in section.links', :key='link.name')
          a(:href='link.href') {{ link.name }}
      .copyright Â© 2018 Jacob Daitzman. All rights reserved.
</template>

<script lang="ts">
import Vue from 'vue'
export default Vue.component('footer-links', {
  data () {
    return {
      sections: [
        {
          name: 'Browse',
          links: [
            { name: 'Home', href: '/' },
            { name: 'Projects', href: 'https://github.com/jdtzmn?tab=repositories' }
          ]
        },
        {
          name: 'Social',
          links: [
            { name: 'Website', href: '/' },
            { name: 'GitHub', href: 'https://github.com/jdtzmn' },
            { name: 'Codepen', href: 'https://codepen.io/jdtzmn' }
          ]
        }
      ]
    }
  }
})
</script>

<style lang="scss" scoped>
@import '~@/styles/colors';
@import '~@/styles/grid';

$background: darken($darkgray, 10%);

.footer {
  // display
  padding-top: 25px;

  // color
  background: linear-gradient($background, darken($background, 5%));
  color: $lightgray;
}

.footer-content {
  // display
  @extend %center-block;
}

.footer-section {
  // display
  margin: 30px 0;

  @include md {
    max-width: 50%;
  }
}

.footer-header {
  // display
  margin-bottom: 10px;

  // font
  font-size: 1.7em;
  font-weight: 500;

  // border
  border-bottom: 2px solid $lightgray;
}

.footer-link > a {
  // color
  color: $lightgray;

  // font
  text-decoration: none;

  // transition
  transition: color 0.1s;

  &:hover {
    // color
    color: $gray;
  }
}

.copyright {
  // color
  color: $lighttext;

  // display
  padding-bottom: 25px;

  // font
  font-size: 0.7em;
  text-align: center;
}
</style>
