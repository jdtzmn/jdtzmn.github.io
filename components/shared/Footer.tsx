import styled from 'styled-components'
import { lighten, darken } from 'polished'
import { Container, Button } from 'components/styled'
import Link from 'next/link'

const Background = styled.footer`
  background: ${({ theme }) => darken(0.025, theme.colors.background)};
  padding: 64px;
  padding-bottom: 48px;
`

const FooterContainer = styled(Container)`
  max-width: 800px;
`

const FooterSections = styled.div`
  display: flex;
  flex-flow: row wrap;
  margin: 0.5em 0 1.5em;
`

const FooterSection = styled.div`
  flex: 1;
  min-width: 7em;
  margin: 0.6em 0;
  &:not(:last-child) {
    margin-right: 2.5em;
  }
`

const FooterSectionTitle = styled.p`
  color: ${({ theme }) =>
    theme.light
      ? lighten(0.2, theme.colors.text)
      : darken(0.05, theme.colors.text)};
  font-size: 0.9em;
  font-weight: 400;
`

const FooterAnchor = styled(Button)`
  color: ${({ theme }) =>
    theme.light
      ? lighten(0.5, theme.colors.text)
      : darken(0.3, theme.colors.text)};
  font-family: 'Inter', sans-serif;
  font-size: 0.9em;
  font-weight: 500;
  padding: 0;
  border: 0;
  margin: 0.75em 0;
`

const Copyright = styled.p`
  text-align: center;
  font-size: 0.7em;
`

type FooterLink = [label: string, link: string]

interface FooterSection {
  name: string
  links: FooterLink[]
}

const footerSections: FooterSection[] = [
  {
    name: 'Links',
    links: [
      ['Home', '/'],
      ['Status', '/status'],
      ['Projects', '/projects'],
    ],
  },
  {
    name: 'Social',
    links: [
      ['GitHub', 'https://github.com/jdtzmn'],
      ['LinkedIn', 'https://linkedin.com/in/jdtzmn'],
    ],
  },
  {
    name: 'Contact',
    links: [['Get in Touch', '/contact']],
  },
]

const Footer = () => {
  const sections = footerSections.map(({ name, links }) => {
    const footerSectionLinks = links.map(([label, link]) => {
      return (
        <Link href={link} passHref key={`footer-${link}`}>
          <FooterAnchor kind="text" as="a">
            {label}
          </FooterAnchor>
        </Link>
      )
    })

    return (
      <FooterSection key={`footer-${name}`}>
        <FooterSectionTitle as="h2">{name}</FooterSectionTitle>
        {footerSectionLinks}
      </FooterSection>
    )
  })

  return (
    <Background role="contentinfo">
      <FooterContainer>
        <FooterSections>{sections}</FooterSections>
      </FooterContainer>
      <Copyright>Â© 2020 Jacob Daitzman. All rights reserved.</Copyright>
    </Background>
  )
}

export default Footer
