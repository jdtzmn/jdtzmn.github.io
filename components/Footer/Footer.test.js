import { shallowMount } from '@vue/test-utils'
import Footer from './Footer'

describe('Footer', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(Footer)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render footer div', () => {
    expect(wrapper.contains('.footer')).toBeTruthy()
  })

  it('should render footer-content div', () => {
    expect(wrapper
      .find('.footer')
      .contains('.footer-content'))
      .toBeTruthy()
  })

  it('should render 2 footer-section divs inside footer-content div', () => {
    expect(wrapper
      .find('.footer-content')
      .findAll('.footer-section'))
      .toHaveLength(2)
  })

  it('should render every footer-section with a footer-header div', () => {
    const sections = wrapper.findAll('.footer-section')
    expect(sections.contains('.footer-header')).toBeTruthy()
    expect(sections.contains('.footer-link')).toBeTruthy()
  })

  it('should render an a element inside every footer-link', () => {
    expect(wrapper.findAll('.footer-link').contains('a')).toBeTruthy()
  })

  it('should render copyright div', () => {
    expect(wrapper.contains('.copyright')).toBeTruthy()
  })
})
