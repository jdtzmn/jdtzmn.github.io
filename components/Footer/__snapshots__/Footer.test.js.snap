// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
<div class="footer">
  <div class="footer-content">
    <div class="footer-section">
      <div class="footer-header">Browse</div>
      <div class="footer-link"><a href="/">Home</a></div>
      <div class="footer-link"><a href="https://github.com/jdtzmn?tab=repositories">Projects</a></div>
    </div>
    <div class="footer-section">
      <div class="footer-header">Social</div>
      <div class="footer-link"><a href="/">Website</a></div>
      <div class="footer-link"><a href="https://github.com/jdtzmn">GitHub</a></div>
      <div class="footer-link"><a href="https://codepen.io/jdtzmn">Codepen</a></div>
    </div>
    <div class="footer-section">
      <div class="footer-header">Blog</div>
      <div class="footer-link"><a href="http://daily.jacob.daitzman.com">100 Days of Coding Problems</a></div>
    </div>
    <div class="copyright">Â© 2018 Jacob Daitzman. All rights reserved.</div>
  </div>
</div>
`;
